frontend/src/entities/configuration/deployment-configuration.tsx

```diff
@@ -0,0 +1,206 @@
+/**
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { setBreadcrumbs } from '../../shared/layout/navigation/navigation.reducer';
+import { useAppDispatch } from '../../config/store';
+import {
+    ExpandableSection,
+    Header,
+    SpaceBetween,
+    Container,
+    FormField,
+    Multiselect,
+    MultiselectProps,
+    SelectProps,
+    Button,
+} from '@cloudscape-design/components';
+import { defaultEnvConfig, envConfig } from '../../shared/model/configuration.model';
+import React, { useEffect, useState } from 'react';
+import { getEnvironmentConfig } from './configuration.service';
+import { getBase } from '../../shared/util/breadcrumb-utils';
+import { useParams } from 'react-router-dom';
+import Condition from '../../modules/condition';
+import axios from '../../shared/util/axios-utils';
+import NotificationService from '../../shared/layout/notification/notification.service';
+
+export function DeploymentConfiguration () {
+    const [config, setEnvConfig] = useState(defaultEnvConfig as envConfig);
```

Comment:
Looks like `defaultEnvConfig` is already an `envConfig`. Is there a reason we have to cast it?